### ISSUE TITLE
Incorrect time complexity - 1361. Validate Binary Tree Nodes

### ISSUE THREAD

**Issue by mdes4214** [LeetCode-Feedback/LeetCode-Feedback#17291]
#### Your LeetCode username

mdes4214

#### Category of the bug
- [ ] Question
- [x] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug

Incorrect time complexity in Approach 3: Union Find.

#### Expected behavior

According to the [reference](https://en.wikipedia.org/wiki/Disjoint-set_data_structure#Time_complexity) shown in the solution. If we only use the path compression without any merge strategy like `rank`, the time complexity of union find will be greater than `O(n)`.

![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/9001974/5c0fd54b-d60c-4dca-b7ac-3163d39a3bc9)

In the worst case of a skewed binary tree, we need `O(n)` to doing `find()`, where `n` is the number of nodes. And the time complexity of all operations will be `O(n * log(n))`

other refs:
- https://stackoverflow.com/questions/2323351/union-find-algorithm-without-union-by-rank-for-disjoint-set-forests-data-structu
- https://cs.stackexchange.com/questions/48649/complexity-of-union-find-with-path-compression-without-rank

**Comment by LC-Pam** [comment]
Hi there,

Thank you for reaching out to us.

 We've relayed this issue to our team to investigate. 

We will reach back to you after we have the results.

Thank you for your support.

**Comment by mdes4214** [comment]
I forgot the fact that the difference of depth between the given parent and child nodes was exactly `1`. So the resulting union find graph would be always depth `1`, and the time complexity of `find()` was surely `O(1)`.

The solution is right. Sorry for the wrong issue.

**Comment by LC-Pam** [comment]
Yes, the structure of this problem is quite different from a regular DSU graph: if a pair of nodes is considered valid for union, only eligible tree root nodes are considered as new children. These new children will always have a rank of 0. Therefore, during the union process, the rank of all nodes will not exceed 1. As for the possibility of nodes having a rank greater than 1, it would be filtered out as required by the problem statement and won't occur.
